src/benchmarks/micro/corefx/System.Linq/Perf.OrderBy.cs

```diff
@@ -0,0 +1,151 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Engines;
+using MicroBenchmarks;
+
+namespace System.Linq.Tests
+{
+    class FavourColourComparer
+        : IComparer<System.Drawing.Color>
+    {
+        public static IComparer<System.Drawing.Color> Instance => new FavourColourComparer();
+        static int TurnIntoValue(System.Drawing.Color x) => (x.R + x.G + x.B) * x.A;
+        public int Compare(System.Drawing.Color lhs, System.Drawing.Color rhs) => TurnIntoValue(lhs).CompareTo(TurnIntoValue(rhs));
+    }
+
+    // Bit of a mix of some value type fields and string
+    public class Person
+    {
+        public string FirstName { get; set; }
+        public string MiddleName { get; set; }
+        public string LastName { get; set; }
+
+        public string FullName => $"{LastName}, {FirstName} {MiddleName[0]}.";
+
+        public DateTime DOB { get; set; }
+        public System.Drawing.Color FavouriteColour { get; set; }
+        public int IQ { get; set; }
+    }
+
+    class PersonData
+    {
+        public static Person[] Generate(int count)
+        {
+            var r = new Random(42); // Same seed for consistent dataset
```

Comment:
thanks for using a const seed! :+1: 

https://github.com/dotnet/performance/blob/master/docs/microbenchmark-design-guidelines.md#always-the-same-input-data